packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -1107,11 +1155,14 @@ abstract class RenderTwoDimensionalViewport extends RenderBox implements RenderA
       _leadingYIndex = math.min(vicinity.yIndex, _leadingYIndex!);
       _trailingYIndex = math.max(vicinity.yIndex, _trailingYIndex!);
     }
-    if (_needsDelegateRebuild || !_children.containsKey(vicinity)) {
+    if (_needsDelegateRebuild || (!_children.containsKey(vicinity) && !_keepAliveBucket.containsKey(vicinity))) {
       invokeLayoutCallback<BoxConstraints>((BoxConstraints _) {
         _childManager._buildChild(vicinity);
       });
     } else {
+      if (_keepAliveBucket.containsKey(vicinity)) {
```

Comment:
is this check needed? I think you can just remove even if it is not in the map?

Reply:
Oh right. Good call.